syntaxes/p.YAML-tmLanguage

```diff
@@ -0,0 +1,148 @@
+# [PackageDev] target_format: plist, ext: tmLanguage
+name: P language
+scopeName: source.p
+fileTypes: [p]
+uuid: 02d90f6f-553b-4a22-a121-a8ddeb6dade4
+
+
+patterns:
+- include: '#comments'
+- name: meta.vardecl
+  match: (\bvar\b)\s*([\w'?]+\b(?!<))?\s*
+  captures:
+    "1":
+      name: keyword.statement.structural
+- name: meta.typedecl
+  match: (\btype\b)\s*([\w'?]+\b(?!<))?\s*
+  captures:
+    "1":
+      name: keyword.statement.structural
+    "2":
+      name: storage.type.other
+#statements
+- name: keyword.statement.structural
+  match: \bvar\b|\bstate\b|\bhot\b|\bcold\b|\bstart\b|\bspec\b|\bcase\b|\bmodule\b|\btest\b|\bmain\b|\bfun\b|\bobserves\b|\bentry\b|\bexit\b|\bwith\b|\bunion\b|\main\b|\btype\b
+- name: keyword.control
+  match: \bif\b|\bgoto\b|\bforeach\b|\bon\b|\bdo\b|\belse\b|\bwhile\b|\breturn\b|\bbreak\b|\bcontinue\b
+- name: keyword.statement.other
+  match: \bassert\b|\bprint\b|\bnew\b|\braise\b|\bsend\b|\bannounce\b
+  
+#expressions
+- name: support.expressions
+  match: \bhalt\b|\bthis\b|\bformat\b|\bnew\b|\bas\b|\bto\b|\bchoose\b\$\b
+- name: support.primitive_functions
+  match: \bsizeof\b|\bkeys\b|\bvalues\b|\bin\b|\bdefault\b
+
+#operators
+- name: keyword.operator.assignment
+  match: =
+- name: keyword.operator.comparison
+  match: <|>|<=|>=
+- name: keyword.operator.arithmetic 
+  match: \-|\+|\*|\%|\/|\!
+
+- name: storage.type
+  match: \b\w*Interface\b|\bI[A-Z]\w*\b
+- include: "#primitives"
+- include: "#functiondeclare"
+- include: "#functioncall"
+- include: "#tuples"
+- include: "#attributes"
+- include: "#othertypes"
+  
+#constants
+- name: constant.language #constant booleans
+  match: \btrue\b|\bfalse\b
+- name: constant.numeric  #constant numbers
+  match: \b([1-9]+[0-9]*|0|[0-9]*\.[0-9]*)\b
+- name: constant.language #strings
+  match: (".*")
+
+- name: keyword.primitive_expressions
+  match: \$
+  captures:
+    '1': {name: keyword.primitive_expressions}
+
+repository:
```

Comment:
add more descriptive comments on the meaning of this section that. will help to understand the code better.